### PR TITLE
desktop alpha

### PR DESCRIPTION
**Pull request by bunsenstraat**
In the main repo yarn, then run yarn serve
In this directory apps/remixdesktop, yarn, then run: yarn start:dev to boot the electron app

### CODE REVIEW

File: apps/remix-ide/src/app/plugins/electron/compilerLoaderPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    if (binRes.status === 200) {
+      response.binList = binRes.data.builds
+    }
+    response.baseURLBin = baseURLBin
+    response.baseURLWasm = baseURLWasm
+
+    this.emit('jsonBinDataLoaded', response)
+  }
+
+  async onActivation(): Promise<void> {
```
Comment (LianaHus) [inline apps/remix-ide/src/app/plugins/electron/compilerLoaderPlugin.ts:50]: do you need this?

File: apps/remix-ide/src/app/plugins/electron/electronConfigPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    super({
+      displayName: 'electronconfig',
+      name: 'electronconfig',
+      description: 'electronconfig',
+    })
+    this.methods = []
+    
+  }
+
+  onActivation(): void {
```
Comment (LianaHus) [inline apps/remix-ide/src/app/plugins/electron/electronConfigPlugin.ts:14]: and this?

File: apps/remix-ide/src/app/plugins/electron/compilerLoaderPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+const methods = ['getJsonBinData']
+
+export class compilerLoaderPlugin extends Plugin {
+  constructor() {
+    super(profile)
+    this.methods = methods
+  }
+
+  async getJsonBinData() {
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/electron/compilerLoaderPlugin.ts:21]: that would be helpful to get a little schema showing the architecture of Electron plugins.
I see there is `ElectronPlugin` but also `ElectronBasePluginClient`, and `getJsonBinData` is declared in this class but not in the `ElectronPlugin` one...

File: apps/remix-ide/src/app/plugins/electron/isoGitPlugin.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    super({
+      displayName: 'isogit',
+      name: 'isogit',
+      description: 'isogit',
+    })
+    this.methods = []
+    
+  }
+
+  async onActivation(): Promise<void> {
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/electron/isoGitPlugin.ts:14]: empty, is this plugin still needed?

File: apps/remix-ide/src/app/plugins/electron/scriptRunnerPlugin.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      const result = await this.call('notification', 'modal', addModuleModal)
+      if (result) {
+        await this.addModule(module)
+      }
+    })
+  }
+
+  async addModule(module: string, version: string = ''): Promise<void> {
+    await this.checkPackageJson()
+    await this.call('scriptRunner', 'yarnAdd', module, version)
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/electron/scriptRunnerPlugin.ts:33]: `yarnAdd` is this a thing?

File: libs/remix-ui/home-tab/src/lib/components/homeTabFeaturedPlugins.tsx [hunk 6]
```diff
@@ -74,8 +74,8 @@ function HomeTabFeaturedPlugins({plugin}: HomeTabFeaturedPluginsProps) {
   }
   const startCookbook = async () => {
     await plugin.appManager.activatePlugin(['cookbookdev'])
-    plugin.verticalIcons.select('cookbookdev')
-    _paq.push(['trackEvent', 'hometabActivate', 'userActivate', 'cookbookdev'])
+    plugin.verticalIcons.select('cookbook.dev')
```
Comment (yann300) [inline libs/remix-ui/home-tab/src/lib/components/homeTabFeaturedPlugins.tsx:77]: I think the name is still `cookbookdev`

Comment (LianaHus) [reply libs/remix-ui/home-tab/src/lib/components/homeTabFeaturedPlugins.tsx:77]: the last one is cookbook.dev

File: apps/remix-ide/src/app/plugins/electron/isoGitPlugin.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    setTimeout(async () => {
+      const version = await this.call('isogit', 'version')
+      if(version){
+      //this.call('terminal', 'log', version)
+      }else{
+      //this.call('terminal', 'log', 'Git is not installed on the system. Using builtin git instead. Performance will be affected. It is better to install git on the system and configure the credentials to connect to GitHub etc.')
```
Comment (LianaHus) [inline apps/remix-ide/src/app/plugins/electron/isoGitPlugin.ts:21]: are those comments needed??

**Review by yann300 — APPROVED** [review]

**Pull request merged**